python/cudf/cudf/core/series.py

```diff
@@ -2746,13 +2745,28 @@ def quantile(
         dtype: float64
         """
 
-        result = self._column.quantile(q, interpolation, exact)
+        return_scalar = is_scalar(q)
+        if return_scalar:
+            np_array_q = np.asarray([float(q)])
+        else:
+            try:
+                np_array_q = np.asarray(q)
+            except TypeError:
+                try:
+                    np_array_q = cudf.core.column.as_column(q).values_host
+                except TypeError:
+                    raise TypeError(
+                        f"q must be a scalar or array-like, got {type(q)}"
+                    )
+        result = self._column.quantile(
+            np_array_q, interpolation, exact, return_scalar=return_scalar
+        )
 
-        if isinstance(q, Number):
+        if return_scalar:
             return result
 
         if quant_index:
-            index = np.asarray(q)
+            index = np_array_q
             if len(self) == 0:
```

Comment:
Why isn't this check happening before we invoke `self._column.quantile`? Also, it looks like `return_scalar` won't be obeyed if `len(self) == 0` and we'll always return a `Series` even if the user requested a single quantile. Does that match pandas behavior? Sorry if this is expanding the scope of this PR, but this issue seems very related. Feel free to push back if you don't think we should touch it in this PR though.

Reply:
you are right, let me refactor this a bit.

Reply:
Fixed this part of the logic